Tests/PactConsumerSwiftTests/AnimalServiceClient.swift

```diff
@@ -9,128 +8,162 @@ public struct Animal {
+        var alligators : [Animal] = []
+        for alligator in jsonResult {
+          alligators.append(Animal(
+            name: alligator["name"] as! String,
+            type: alligator["type"] as! String,
+            dob: alligator["dateOfBirth"] as? String,
+            legs: alligator["legs"] as? Int))
         }
+        response(alligators)
       }
     }
   }
 
   open func eat(animal: String, success: @escaping () -> Void, error: @escaping (Int) -> Void) {
-    Alamofire.request("\(baseUrl)/alligator/eat", method: .patch, parameters: [ "type" : animal ], encoding: JSONEncoding.default)
-    .responseString { (response) in
-      if response.result.isFailure {
-        error(response.response!.statusCode)
+    self.performRequest("\(baseUrl)/alligator/eat", method: "patch", parameters: ["type": animal], isJsonResponse: false) { jsonObject, nsError in
+      if let localErr = nsError {
+        error(localErr.code)
       } else {
         success()
       }
     }
   }
 
   open func wontEat(animal: String, success: @escaping () -> Void, error: @escaping (Int) -> Void) {
-    Alamofire.request("\(baseUrl)/alligator/eat", method: .delete, parameters: [ "type" : animal ], encoding: JSONEncoding.default)
-    .responseJSON { (response) in
-      if response.result.isFailure {
-        error(response.response!.statusCode)
+    self.performRequest("\(baseUrl)/alligator/eat", method: "delete", parameters: ["type": animal]) { jsonObject, nsError in
+      if let localErr = nsError {
+        error(localErr.code)
       } else {
         success()
       }
     }
   }
 
   open func eats(_ success: @escaping ([Animal]) -> Void) {
-    Alamofire.request("\(baseUrl)/alligator/eat")
-    .responseJSON { (response) in
-      if response.result.isSuccess {
-        if let jsonResult = response.result.value as? Array<Dictionary<String, AnyObject>> {
-          var animals: [Animal] = []
-          for alligator in jsonResult {
-            animals.append(Animal(
-              name: alligator["name"] as! String,
-              type: alligator["type"] as! String,
-              dob: alligator["dateOfBirth"] as? String,
-              legs: alligator["legs"] as? Int))
+    self.performRequest("\(baseUrl)/alligator/eat") { jsonObject, nsError in
+      if let jsonResult = jsonObject as? Array<Dictionary<String, AnyObject>> {
+        var animals: [Animal] = []
+        for alligator in jsonResult {
+          animals.append(Animal(
+            name: alligator["name"] as! String,
+            type: alligator["type"] as! String,
+            dob: alligator["dateOfBirth"] as? String,
+            legs: alligator["legs"] as? Int))
+        }
+        success(animals)
+      }
+    }
+  }
+
+  private func performRequest(_ urlString: String, headers: [String: String]? = nil, method: String = "get", parameters: [String: String]? = nil, isJsonResponse: Bool = true, completionHandler: @escaping (_ jsonObject: Any?, _ error: NSError?) -> Void) {
```

Comment:
There are so many arguments in this function, please consider the following suggestion for readability:
```
performRequest(
  _ urlString: String, 
  headers: [String: String]? = nil, 
  method: String = "get", 
  parameters: [String: String]? = nil, 
  isJsonResponse: Bool = true, 
  completionHandler: @escaping (_ jsonObject: Any?, _ error: NSError?) -> Void
) {
...
```

Am also a bit surprised _SwiftLint_ has not picked up this `line_length` violation.